docs/technical-overview.md

```diff
@@ -0,0 +1,267 @@
+
+Each fixture belongs to a [manufacturer](#manufacturers) (many-to-one relationship). A manufacturer is the vendor or brand of the fixture.
+
+The fixtures are saved as [JSON](http://www.json.org/) files at `fixtures/<manufacturer-key>/<fixture-key>.json`. The fixture key is only defined by the filename. See [details about the fixture JSON format](TODO).
+
+The JSON fixture data is parsed and processed using our [model](#fixture-model).
+
+## Manufacturers
+*[⬆️ Back to top](#technical-overview)*
+
+A manufacturer is a [fixture](#fixtures) vendor or brand. Each fixture belongs to a manufacturer (0-n fixtures to 1 manufacturer relationship).
+
+All used manufacturers must be defined in `fixtures/manufacturers.json` in order that their unique keys can be registered. All fixtures of a manufacturer are saved in the `fixtures/<manufacturer-key>/` directory.
+
+We also store (optional) additional manufacturer data like comment, website or [RDM](TODO) data.
+
+## Fixture model
+*[⬆️ Back to top](#technical-overview)*
+
+Instead of parsing a [fixture](#fixtures)'s JSON data directly, it is recommended to use the model. We developed it to easier handle complicated fixture features like fine/switching channels.
+
+The model uses [ES2015 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) to represent the fixtures. E. g., `Fixture.fromRepository('cameo', 'nanospot-120')` returns an instance of the `Fixture` class, instantiated with the specified fixture's data. These objects have several convenient properties that allow an easy usage of the fixture data in [plugins](#plugins), [UI](#ui--website) and more.
+
+All model classes (like `Fixture`, `Manufacturer`, `Physical`, `Channel` or `Range`) are located in the `lib/model/` directory. When using the model, it should be enough importing the `Fixture` module; instances of other classes are returned by the fixture's properties:
+
+```js
+const Fixture = require('lib/model/Fixture.js');
+
+const myFix = Fixture.fromRepository('cameo', 'nanospot-120'); // instanceof Fixture
+
+const physicalData = myFix.physical; // instanceof Physical
+const panFine = myFix.getChannelByKey('Pan fine'); // instanceof FineChannel
+
+if (panFine.coarseChannel.hasHighlightValue) {
+  console.log(`Hightlight at ${panFine.coarseChannel.highlightValue}`)
+}
+```
+
+Model properties are always implemented using getters and setters. To store data, we use backing fields (prefixed with underscore, like `_jsonObject`) that should never be invoked from outside. Properties that need further computation or create other objects should be cached in the internal `_cache` object. Avoid returning `undefined` under all circumstances! To achieve this, use defaults for optional properties – if the default is not `null`, it's good practice providing a `hasXY` boolean.
```

Comment:
* What are backing fields? I don't know this name.
* Only functions can be invoked, properties (and thus also getters and setters) can only be accessed.
* "it is good practice to provide"